pillar.example

```diff
@@ -1,18 +1,18 @@
 postgres:
-  # Set True to configure upstream postgresql.org repository for YUM or APT
+  # Set True to configure upstream postgresql.org repository for YUM/APT/ZYPP
   use_upstream_repo: False
   # Version to install from upstream repository
-  version: '9.3'
+  version: '9.6'
 
   # These are Debian/Ubuntu specific package names
-  pkg: 'postgresql-9.3'
-  pkg_client: 'postgresql-client-9.3'
+  pkg: 'postgresql-9.6'
+  pkg_client: 'postgresql-client-9.6'
 
   # Additional packages to install with PostgreSQL server,
   # this should be in a list format
-  pkgs_extra:
-    - postgresql-contrib
-    - postgresql-plpython
+  #pkgs_extra:
+  #  - postgresql-contrib
```

Comment:
Regarding your comment above. If we're talking about Ubuntu, `postgresql-contrib` is definitely optional package, it does not contain `initdb` binary.
https://packages.ubuntu.com/trusty/amd64/postgresql-contrib-9.3/filelist
https://packages.ubuntu.com/xenial/amd64/postgresql-contrib-9.5/filelist

I think the issue with missing `initdb` lies somewhere else, maybe in our beloved alternatives?
So commenting out these extras is a good thing.

Reply:
Yeah, `postgres-contrib` creates alternatives. The formula requires initdb in PATH (see below) or state failures occur.
```
./postgres/defaults.yaml:    command: initdb --pgdata=/var/lib/pgsql/data
./postgres/osmap.yaml:    command: initdb -D /var/lib/postgres/data
./postgres/osmap.yaml:    command: initdb --pgdata='{{ data_dir }}'
./postgres/osmap.yaml:    - initdb
./postgres/osmap.yaml:    command: initdb --pgdata='{{ lib_dir }}'
./postgres/osmap.yaml:    command: initdb -D /Users/{{ repo.user }}/Library/AppSupport/postgres_{{ repo.use_upstream_repo }}
```

Do you know how `initdb` gets added to PATH without `postgres-contrib` package in this formula?  The redhat os mapping invokes saltstack to create alternatives, so the package must have been missing when that feature was missing.

Back to the specific comment - I would prefer to remove pkgs_extra pillar example, because it corrupted my installation once or twice, this is bad onboarding experience for new users. Otherwise commenting is definitely a good thing - I agree and will implement.

Reply:
This is because of the `postgres-contrib` is just a metapackage that *also* install a full PostgreSQL server + client of particular version available in the distro by dependencies. And there we have all the alternatives installation code in the downstream.
So if you have a wrong or empty value in the `pkg` Pillar value, the `postgres-contrib` just does the right thing.

This is just a very handy coincidence, that does not make `postgres-contrib` package mandatory.

Reply:
Okay. In this PR I explicitly fix the INITDB problem (#160) via salt.alternatives.install. As documented in #160, upstream packages do not add initdb to PATH by design. 

Here is my verified solution - I am unaware of any postgres-defined solution.
```
...
 +  prepare_cluster:
 +    #Full path needed as initdb is NOT 'cross version compatible' binary
 +    command: /usr/pgsql-{{ release }}/bin/initdb --pgdata='{{ lib_dir }}'
 +    test: test -f '{{ lib_dir }}/PG_VERSION'
 +
 +  # Handles etc alternatives for 'cross version compatible' binaries
 +  pkgs_extra: ['postgresql{{ release }}-contrib',]
 +
 +{% else %}
```
The solution fails when pillars overrides the verified os mapping values. 

If this solution is acceptable to you, let me know.

Reply:
Yeah. That is a separate issue appearing exclusively when installing from upstream.
We will handle this. Thank you.